src/errors.ts

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export namespace unstable {
+    /**
+     * Append the old error message to the new one and keep its stack trace.
+     * Example:
+     *     throw wrapError(e, HighLevelError, "This error is more specific");
+     */
+    export function wrapError<T extends Error>(
```

Comment:
Do we have to flag this renamed function as a breaking change or can it not be used by external software?

Reply:
This release is going to be breaking (2.0) so I'm not concerned about an alias, but we should call out that it's changing name.

Reply:
bth. I would be annoyed by a lib making this seemingly trivial name change.

It would be a bit nicer with some grace period where the old function is an alias and console log's that it's going to be deprecated on first usage.

Reply:
If the parent project doesn't use TypeScript this will likely fail on runtime. And, given this is related to error logging, it may become though to see if a component or the error reporting failed.

Reply:
I'll add a warning.